### PR TITLE
Add CodeQL GH Action

### PR DESCRIPTION
**Pull request by hawflau**
#### Why is this change necessary?
Enable Code Scan through GH Action

#### Mandatory Checklist
**PRs will only be reviewed after checklist is complete**

- [ ] Add input/output [type hints](https://docs.python.org/3/library/typing.html) to new functions/methods
- [ ] Write design document if needed ([Do I need to write a design document?](https://github.com/aws/aws-sam-cli/blob/develop/DEVELOPMENT_GUIDE.md#design-document))
- [ ] Write/update unit tests
- [ ] Write/update integration tests
- [ ] Write/update functional tests if needed
- [ ] `make pr` passes
- [ ] `make update-reproducible-reqs` if dependencies were changed
- [ ] Write documentation

By submitting this pull request, I confirm that my contribution is made under the terms of the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0).

### CODE REVIEW

File: samcli/lib/pipeline/bootstrap/stage.py [hunk 1]
```diff
@@ -201,6 +201,10 @@ def generate_thumbprint(oidc_provider_url: Optional[str]) -> Optional[str]:
         # Create an IPV4 socket and use TLS for the SSL connection
         address = (url_for_certificate, 443)
         ctx = SSL.Context(SSL.TLS_METHOD)
+        ctx.set_options(SSL.OP_NO_TLSv1)
```
Comment (sriram-mv) [inline samcli/lib/pipeline/bootstrap/stage.py:204]: should we verify this with unit tests?

**Review by sriram-mv — COMMENTED** [review]
Is this going to be a gating workflow, preventing PR merges? What kind of actions does the unsuccessful run of codeql action entail on us?

**Review by sriram-mv — CHANGES_REQUESTED** [review]
Changing my review to reflect requesting changes.

**Comment by hawflau** [comment]
@sriram-mv 

> Is this going to be a gating workflow, preventing PR merges? What kind of actions does the unsuccessful run of codeql action entail on us?

Yes, it's run on each PR. Failure of this action should prevent the PR from being merged. And the PR author should fix me. 

The action will also run on schedule, because existing code might be subjected to newly discovered vulnerabilities. We will get notified and also can check from the "Security" tab in the repo.

File: samcli/lib/pipeline/bootstrap/stage.py (hunk 1 above)
Comment (hawflau) [reply samcli/lib/pipeline/bootstrap/stage.py]: Added

**Review by torresxb1 — APPROVED** [review]

**Pull request merged**